docs/03-concepts/01-actor-lifecycle.mdx

```diff
@@ -0,0 +1,115 @@
+tearing down the event manager and the storage client.
+There is also the [`Actor.fail()`](../../reference/class/Actor#fail) method, which exits the actor while marking it as failed.
+
+```python title="src/main.py"
+from apify import Actor
+from apify.consts import ActorExitCodes
+
+async def main():
+    await Actor.init()
+    try:
+        print('Actor input:', await Actor.get_input())
+        await Actor.set_value('OUTPUT', 'Hello, world!')
+        raise RuntimeError('Ouch!')
+        await Actor.exit()
+    except Exception as e:
+        print('Error while running actor', e)
+        await Actor.fail(exit_code=ActorExitCodes.ERROR_USER_FUNCTION_THREW, exception=e)
+```
+
+### Context manager
+
+So that you don't have to call the lifecycle methods manually, the [`Actor`](../../reference/class/Actor) class provides a context manager,
+which calls the [`Actor.init()`](../../reference/class/Actor#init) method on enter,
+the [`Actor.exit()`](../../reference/class/Actor#exit) method on a clean exit,
+and the [`Actor.fail()`](../../reference/class/Actor#fail) method when there is an exception during the run of the actor.
+
+This is the recommended way to work with the `Actor` class.
+
+```python title="src/main.py"
+from apify import Actor
+
+async def main():
+    async with Actor:
+        print('Actor input:', await Actor.get_input())
+        await Actor.set_value('OUTPUT', 'Hello, world!')
+        raise RuntimeError('Ouch!')
+```
+
+### Main function
+
+Another option is to pass a function to the actor via the [`Actor.main(main_func)`](../../reference/class/Actor#main) method,
+which causes the actor to initialize, run the main function, and exit, catching any runtime errors in the passed function.
+
+```python title="src/main.py"
+from apify import Actor
+
+async def actor_main_func():
+    print('Actor input:', await Actor.get_input())
+    await Actor.set_value('OUTPUT', 'Hello, world!')
+    raise RuntimeError('Ouch!')
+
+async def main():
+    await Actor.main(actor_main_func)
+```
+
+## Actor status message
+
+To inform you or the users running your actors about the progress of their runs,
+you can set the status message for the run, which will then be visible in the run detail in the Apify Console,
+or accessible through the Apify API.
+
+To set the status message for the actor run, you can use the [`Actor.set_status_message()`](../../reference/class/Actor#set_status_message) method.
+
+```python title="src/main.py"
+from apify import Actor
+
+async def main():
+    async with Actor:
+        await Actor.set_status_message('Here we go!')
+        ...
+        await Actor.set_status_message('So far so good...')
+        ...
+        await Actor.set_status_message('Steady as she goes...')
+        ...
+        await Actor.set_status_message('Almost there...')
+        ...
+        await Actor.set_status_message('Phew! That was not that hard!')
+```
+
+## Rebooting an actor
```

Comment:
Should be reboot under lifecycle methods as well.
![image](https://user-images.githubusercontent.com/11837643/223150705-b8273df5-c448-4c66-9c35-8b0462172d1a.png)

Reply:
Good point, I'll reshuffle this a bit.